merkletree/proof_test.go

```diff
@@ -53,6 +53,13 @@ func TestVerifyProof(t *testing.T) {
 
 	// proof of inclusion
 	proof := m.Get(index3)
+	if proof.Leaf.Value() == nil {
+		t.Fatal("Expect returned leaf's value is not nil")
+	}
+	// ensure this is a proof of inclusion
+	if !bytes.Equal(proof.LookupIndex, proof.Leaf.Index()) {
```

Comment:
Yup. I know the tests are already checking both conditions individually. I know this is nitpicky, but what I'm trying to say is that, really specifically, ``proof.LookupIndex == proof.Leaf.Index()`` alone doesn't imply it's a proof of inclusion for the name we're interested in.